app/services/wordfilter_service.rb

```diff
@@ -0,0 +1,40 @@
+class WordfilterService
+  def initialize(location, text)
+    @location = location
+    @text = text
+  end
+
+  def reject?
+    wordfilters[:reject].present?
+  end
+
+  def hide?
+    wordfilters[:hide].present?
+  end
+
+  def report?
+    wordfilters[:report].present?
+  end
+
+  def censor?
+    wordfilters[:censor].present?
+  end
+
+  def censored_text
+    wordfilters[:censor].inject(@text) do |text, wordfilter|
+      # Convert SQL LIKE roughly into Regex
+      pattern = if wordfilter.regex_enabled? then wordfilter.pattern
+                else wordfilter.pattern.gsub('.', '\.').gsub('%', '.*').tr('_', '.')
+                end
```

Comment:
🤮

Reply:
Lmao right? It's nasty but I want to make shit work the same across db and rails 😭